### ISSUE TITLE
Kong start does not create the full contents of the prefix directory. Does not start properly

### ISSUE THREAD

**Issue by opensauce17** [Kong/kong#3485]
When starting kong the contents of the prefix folder only has : 
**logs  nginx.conf  pids  ssl**

Subsequently kong does not start. I have found that this is inconsistent, but it fails to create the contents of the directory more often than not. 

**Kong version :** 0.13.1

**Operating System :**  Ubuntu 16.04.4 LTS

**Kong.conf :** 
proxy_listen = 0.0.0.0:80, 0.0.0.0:443 ssl
admin_listen = 0.0.0.0:8001, 0.0.0.0:8444 ssl
pg_host = <PG IP ADDRESS>
pg_port = 5432
pg_user = postgres
pg_password = <PASSWORD>
pg_database = kong

**Note : I have ommitted DB details i.e host and password.**

**Kong Verbose Start :**

root@kong01:/usr/local# kong start -c /opt/kong/kong.conf --vv
2018/05/24 13:38:34 [verbose] Kong: 0.13.1
2018/05/24 13:38:34 [debug] ngx_lua: 10011
2018/05/24 13:38:34 [debug] nginx: 1013006
2018/05/24 13:38:34 [debug] Lua: LuaJIT 2.1.0-beta3
2018/05/24 13:38:34 [verbose] reading config file at /opt/kong/kong.conf
2018/05/24 13:38:34 [debug] admin_access_log = "logs/admin_access.log"
2018/05/24 13:38:34 [debug] admin_error_log = "logs/error.log"
2018/05/24 13:38:34 [debug] admin_http2 = "off"
2018/05/24 13:38:34 [debug] admin_listen = {"0.0.0.0:8001","0.0.0.0:8444"}
2018/05/24 13:38:34 [debug] admin_listen_ssl = "127.0.0.1:8444"
2018/05/24 13:38:34 [debug] admin_ssl = "on"
2018/05/24 13:38:34 [debug] anonymous_reports = true
2018/05/24 13:38:34 [debug] cassandra_consistency = "ONE"
2018/05/24 13:38:34 [debug] cassandra_contact_points = {"127.0.0.1"}
2018/05/24 13:38:34 [debug] cassandra_data_centers = {"dc1:2","dc2:3"}
2018/05/24 13:38:34 [debug] cassandra_keyspace = "kong"
2018/05/24 13:38:34 [debug] cassandra_lb_policy = "RoundRobin"
2018/05/24 13:38:34 [debug] cassandra_port = 9042
2018/05/24 13:38:34 [debug] cassandra_repl_factor = 1
2018/05/24 13:38:34 [debug] cassandra_repl_strategy = "SimpleStrategy"
2018/05/24 13:38:34 [debug] cassandra_schema_consensus_timeout = 10000
2018/05/24 13:38:34 [debug] cassandra_ssl = false
2018/05/24 13:38:34 [debug] cassandra_ssl_verify = false
2018/05/24 13:38:34 [debug] cassandra_timeout = 5000
2018/05/24 13:38:34 [debug] cassandra_username = "kong"
2018/05/24 13:38:34 [debug] client_body_buffer_size = "8k"
2018/05/24 13:38:34 [debug] client_max_body_size = "0"
2018/05/24 13:38:34 [debug] client_ssl = false
2018/05/24 13:38:34 [debug] custom_plugins = {}
2018/05/24 13:38:34 [debug] database = "postgres"
2018/05/24 13:38:34 [debug] db_cache_ttl = 3600
2018/05/24 13:38:34 [debug] db_update_frequency = 5
2018/05/24 13:38:34 [debug] db_update_propagation = 0
2018/05/24 13:38:34 [debug] dns_error_ttl = 1
2018/05/24 13:38:34 [debug] dns_hostsfile = "/etc/hosts"
2018/05/24 13:38:34 [debug] dns_no_sync = false
2018/05/24 13:38:34 [debug] dns_not_found_ttl = 30
2018/05/24 13:38:34 [debug] dns_order = {"LAST","SRV","A","CNAME"}
2018/05/24 13:38:34 [debug] dns_resolver = {}
2018/05/24 13:38:34 [debug] dns_stale_ttl = 4
2018/05/24 13:38:34 [debug] error_default_type = "text/html"
2018/05/24 13:38:34 [debug] http2 = "off"
2018/05/24 13:38:34 [debug] latency_tokens = true
2018/05/24 13:38:34 [debug] log_level = "notice"
2018/05/24 13:38:34 [debug] lua_package_cpath = ""
2018/05/24 13:38:34 [debug] lua_package_path = "./?.lua;./?/init.lua;"
2018/05/24 13:38:34 [debug] lua_socket_pool_size = 30
2018/05/24 13:38:34 [debug] lua_ssl_verify_depth = 1
2018/05/24 13:38:34 [debug] mem_cache_size = "128m"
2018/05/24 13:38:34 [debug] nginx_daemon = "on"
2018/05/24 13:38:34 [debug] nginx_optimizations = true
2018/05/24 13:38:34 [debug] nginx_user = "nobody nobody"
2018/05/24 13:38:34 [debug] nginx_worker_processes = "auto"
2018/05/24 13:38:34 [debug] pg_database = "kong"
2018/05/24 13:38:34 [debug] pg_host = "10.251.3.18"
2018/05/24 13:38:34 [debug] pg_password = "******"
2018/05/24 13:38:34 [debug] pg_port = 5432
2018/05/24 13:38:34 [debug] pg_ssl = false
2018/05/24 13:38:34 [debug] pg_ssl_verify = false
2018/05/24 13:38:34 [debug] pg_user = "postgres"
2018/05/24 13:38:34 [debug] prefix = "/usr/local/kong/"
2018/05/24 13:38:34 [debug] proxy_access_log = "logs/access.log"
2018/05/24 13:38:34 [debug] proxy_error_log = "logs/error.log"
2018/05/24 13:38:34 [debug] proxy_listen = {"0.0.0.0:80","0.0.0.0:443 ssl"}
2018/05/24 13:38:34 [debug] proxy_listen_ssl = "0.0.0.0:8443"
2018/05/24 13:38:34 [debug] real_ip_header = "X-Real-IP"
2018/05/24 13:38:34 [debug] real_ip_recursive = "off"
2018/05/24 13:38:34 [debug] server_tokens = true
2018/05/24 13:38:34 [debug] ssl = "on"
2018/05/24 13:38:34 [debug] ssl_cipher_suite = "modern"
2018/05/24 13:38:34 [debug] ssl_ciphers = "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256"
2018/05/24 13:38:34 [debug] trusted_ips = {}
2018/05/24 13:38:34 [debug] upstream_keepalive = 60
2018/05/24 13:38:34 [verbose] prefix in use: /usr/local/kong
2018/05/24 13:38:35 [verbose] preparing nginx prefix directory at /usr/local/kong
2018/05/24 13:38:35 [info] prefix directory /usr/local/kong not found, trying to create it
2018/05/24 13:38:35 [verbose] SSL enabled, no custom certificate set: using default certificate
2018/05/24 13:38:35 [verbose] generating default SSL certificate and key
2018/05/24 13:38:35 [warn] ulimit is currently set to "1024". For better performance set it to at least "4096" using "ulimit -n"
2018/05/24 13:38:35 [verbose] could not start Kong, stopping services
2018/05/24 13:38:35 [verbose] stopped services
Error:
/usr/local/share/lua/5.1/kong/cmd/start.lua:68: /usr/local/share/lua/5.1/kong/cmd/utils/prefix_handler.lua:251: invalid replacement value (a table)
stack traceback:
	[C]: in function 'error'
	/usr/local/share/lua/5.1/kong/cmd/start.lua:68: in function 'cmd_exec'
	/usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:87>
	[C]: in function 'xpcall'
	/usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:44>
	/opt/kong/bin/kong:7: in function 'file_gen'
	init_worker_by_lua:49: in function <init_worker_by_lua:47>
	[C]: in function 'xpcall'
	init_worker_by_lua:56: in function <init_worker_by_lua:54>

root@kong01:/usr/local# ls /usr/local/kong/
logs  nginx.conf  pids  ssl

**Comment by p0pr0ck5** [comment]
@opensauce17 curious, is this in a machine with a fresh install of 0.13.1, or an upgrade from an earlier version of Kong such as 0.12? This sounds similar to #2319 in which a replacement value (of type `table`) is passed into `gsub`; if the `nginx_kong.lua` template wasn't upgraded from 0.12 this might result in a failed substitution and the error reported?

**Comment by opensauce17** [comment]
Hi,

Yes, I think you are totally right! Thanks for the insight. 

I upgraded to 0.13.0 and had my automation tool dropping my 0.12.0 templates. Im still unfamiliar with the changes in 0.13.0. I've modified my automation to strictly use the versions ( kong, openresty, luarocks) that I am comfortable with.

I'll spend some time looking at 0.13.0 and consider upgrading to it at some point. 

This issue can be considered close.
